docs/UserGuide.md

```diff
@@ -266,69 +262,150 @@ Example:
+versions ago, assuming you performed at least 2 <em>undoable</em> commands previously!
+* When you perform an <em>undoable</em> command, it will be stacked on top of the
+previous <em>undoable</em> command in the stack. The `undo` command will undo the first <em>undoable</em> command at the
+top of the stack.
+
+* If you have not executed an <em>undoable</em> command previously, the WorkBook will remain in its current version
+and return an error message: "No previous changes to undo!". 
+
+![UndoFailure](images/UndoFailure.png)
 
-Undoable commands: commands that modify the WorkBook's content (`add`, `delete`, `edit` and `clear`)
 
 Examples:
-* `delete 1`
+* `add c/Meta r/Frontend Engineer s/Application Sent d/29-Oct-2022 12:00 e/hrmonkey@example.com` followed by `undo` will
+return the WorkBook version without the added internship.
+* `edit 1 s/Behavioural Interview e/hr@meta.com` followed by `clear` and then `undo` will undo the 
+clearing of the Workbook. The edited stage and email address of the first internship will remain in the 
+returned WorkBook version. 
+
+Workbook version before `undo`:
+![BeforeUndoStack](images/BeforeUndoStack.png)
+
+Workbook version after `undo`:
+![AfterUndoStack](images/AfterUndoStack.png)
+
+
+* `delete 2` followed by `list` and then `undo` will restore the WorkBook to its version prior to the 
+`delete 2` command as the `list` command is not an <em>undoable</em> command and does not create a new
+version of the WorkBook.
+* `edit 2 l/golang t/` followed by `delete 1` and then 2 consecutive `undo` commands will first restore the
+WorkBook to its version before the `delete 1` command, and then the version before the `edit` command.
+
+Before executing command: `edit 2 l/golang t/`
+![BeforeEdit](images/BeforeEdit.png)
+
+After executing command: `edit 2 l/golang t/`
+![AfterEdit](images/AfterEdit.png)
 
-    `list`
+After executing command: `delete 1`
+![AfterDelete](images/AfterDelete.png)
 
-    `undo` (reverses the `delete 1` command)
+After executing command: `undo`
+![AfterUndo1](images/AfterUndo1.png)
 
+After executing command: `undo`
+![AfterUndo2](images/AfterUndo2.png)
 
-* `find google`
 
-  `list`
 
-  `undo` 
+<div markdown="block" class="alert alert-info">
 
-    The `undo` command fails as there are no undoable commands executed previously.
+**:information_source: Note:**<br>
+You can only undo <em>undoable</em> commands!
+* `add`
+* `edit`
+* `delete`
+* `clear`
+* `redo`
+</div>
 
-    
-* `delete 1`
-    
-    `clear`
+<div markdown="block" class="alert alert-success">
 
-    `undo` (reverses the `clear` command)
+**:bulb: Tip:** <br>
+* If you have undone your previous undoable command by mistake, fret not as you can easily redo this 
+to reapply the initial changes!
+* Performing multiple `undo` commands may confuse you as to which <em>undoable</em> command you are
+undoing and how your WorkBook currently looks like!
```

Comment:
i think dunnid include this as it may look like users are expecting that they can change states manually, ygwim right, what do u think?